lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -0,0 +1,317 @@
+    else:
+        # dashboard does not exists : do nothing
+        result = {'msg': "Dashboard {} does not exists".format(data['slug']),
+                  'changed': False,
+                  'slug': data['slug']}
+
+    return result
+
+
+def grafana_export_dashboard(module, data):
+
+    # define http headers
+    headers = {'content-type': 'application/json'}
+    if 'grafana_api_key' in data and data['grafana_api_key']:
+        headers['Authorization'] = "Bearer {}".format(data['grafana_api_key'])
+    else:
+        auth = base64.encodestring('%s:%s' % (data['grafana_user'], data['grafana_password'])).replace('\n', '')
+        headers['Authorization'] = 'Basic %s' % auth
+        grafana_switch_organisation(module, data['grafana_url'], data['org_id'], headers)
+
+    # test if dashboard already exists
+    dashboard_exists, dashboard = grafana_dashboard_exists(module, data['grafana_url'], data['slug'], headers=headers)
+
+    if dashboard_exists is True:
+        with open(data['path'], 'w') as f:
+            f.write(json.dumps(dashboard))
+        result = {'msg': "Dashboard {} exported to {}".format(data['slug'], data['path']),
+                  'slug': data['slug'],
+                  'changed': True}
+    else:
+        result = {'msg': "Dashboard {} does not exists".format(data['slug']),
+                  'slug': data['slug'],
+                  'changed': False}
+
+    return result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(choices=['present', 'absent', 'export'],
+                       default='present'),
+            grafana_url=dict(required=True),
+            grafana_user=dict(default='admin'),
+            grafana_password=dict(default='admin', no_log=True),
+            grafana_api_key=dict(type='str', no_log=True),
+            org_id=dict(default=1, type='int'),
+            slug=dict(type='str'),
+            path=dict(type='str'),
+            overwrite=dict(type='bool', default=False),
+            message=dict(type='str'),
+            validate_certs=dict(type='bool')
+        ),
+        supports_check_mode=True
+    )
+
+    try:
+        if module.params['state'] == 'present':
+            result = grafana_create_dashboard(module, module.params)
+        elif module.params['state'] == 'absent':
+            result = grafana_delete_dashboard(module, module.params)
+        else:
+            result = grafana_export_dashboard(module, module.params)
+    except GrafanaAPIException as e:
+        module.fail_json(
+            failed=True,
+            msg="error : {} ".format(e)
+        )
+        return
+    except GrafanaMalformedJson as e:
+        module.fail_json(
+            failed=True,
+            msg="error : json file does not contain a meta section with a slug parameter, or you did'nt specify the slug parameter"
+        )
+        return
+
+    assert "changed" in result
+    assert "msg" in result
```

Comment:
Please remove uses of `assert` from the code.  See https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/dev_guide/testing/sanity/no-assert.rst